src/liquid_parquet/src/cache/store.rs

```diff
@@ -234,7 +275,54 @@ impl CacheStore {
         self.budget.add_used_disk_bytes(bytes.len());
     }
 
-    pub(super) fn insert(&self, entry_id: CacheEntryID, mut batch_to_cache: CachedBatch) {
+    fn write_to_disk_async(&self, entry_id: &CacheEntryID, liquid_array: &LiquidArrayRef) {
+        let bytes = liquid_array.to_bytes();
+        let file_path = entry_id.on_disk_path(self.config.cache_root_dir());
+        let disk_size = bytes.len();
+        tokio::task::block_in_place(|| {
+            let handle = tokio::runtime::Handle::current();
+            handle.block_on( async {
+                tokio::fs::write(file_path, bytes).await.expect("Failed to write to disk");
+            })
+        });        
+        self.budget.add_used_disk_bytes(disk_size);
+    }
+
+    fn write_to_disk_blocking_uring(&self, entry_id: &CacheEntryID, liquid_array: &LiquidArrayRef) {
+        let bytes = liquid_array.to_bytes();
+        let file_path = entry_id.on_disk_path(self.config.cache_root_dir());
+        let disk_size = bytes.len();
+            
+        tokio::task::block_in_place(|| {
+            tokio_uring::start(async {
+                let file = tokio_uring::fs::File::create(file_path).await;
+                if let Ok(file) = file {
+                    let (res, _) = file.write_all_at(bytes, 0).await;
+                    if res.is_err() {
+                        log::error!("Failed to write to file: {}", res.err().unwrap());
+                    }
+                    self.budget.add_used_disk_bytes(disk_size);
+                    file.close().await.expect("Failed to close file");
+                    
+                } else {
+                    panic!("Failed to create file with tokio-uring");
+                }
+        })});
+    }
+
+    fn write_to_disk_threadpool_uring(self: &Arc<Self>, entry_id: CacheEntryID, liquid_array: LiquidArrayRef) {
+        tokio::task::block_in_place(|| 
+            {
+                let handle = IO_URING_THREAD_POOL.handle();
+                let store_arc = Arc::clone(self);
+                handle.block_on(async move {
```

Comment:
For the thread pool implementation, I suggest we use the lower level library: https://github.com/tokio-rs/io-uring
to manage the sub/comp queues.

Basically, start a thread pool, create a message passing worker to wait for other threads to pass a new io task.
Then manage the issue the low_level io_uring opcode, polling the kernel (or wait for kernel interrupt), once we get the io content, send the message back to the issuing thread.